test/api_test.py

```diff
@@ -135,3 +135,14 @@ def test_create_command(self):
 
     self.assertIsInstance(custom_command, CustomCommand)
     self.assertEqual(custom_command.command, 'helloWorld')
+
+  def test_convert_units(self):
+    """
+    Converting between iota standard units.
+    """
+    api = Iota(MockAdapter())
+
+    self.assertEqual(api.convert_units('1.618 Mi', 'i'), 1618000)
+    self.assertEqual(api.convert_units('42 i', 'Ki'), 0.042)
+    with self.assertRaises(ValueError):
+      api.convert_units('3.141596 Xi', 'Pi')
```

Comment:
`convert_value_to_standard_unit` deserves its own test case.  This makes it easier to test just that function in isolation from the rest of the library.

Also, I would like each scenario split into its own test (e.g., create `test_convert_to_smaller_unit`, `test_convert_to_bigger_unit`, `test_error_invalid_unit`, and so on).  This makes it easier to understand what is being tested, and if any assertion fails, it doesn't prevent the other assertions from being checked.

Lastly, there are a few more tests that need to be added for full coverage:

- Input is the wrong type (e.g., `[1.618, 'Mi']`).
- Input is already a float value (e.g., `1.618`).  Probably same result as the previous one.
- Input is empty (i.e., `''`).
- Input has too few parts (e.g., `'1.618`).
- Input is missing space (e.g., `'1.618Mi'`).  We can treat this as an error.
- Input has too many parts (e.g., `'1 618 Mi'`).
- Result includes fractional IOTAS (e.g., `'1.6189 Ki'`).  This is not an error, just using unit tests to document that this behavior is allowed.
- Input is negative (e.g., `'-1.618 Mi'`).
- Input uses different case (e.g., `'1.618 mI'`).  This should probably be an error... probably.